src/Advection/vector_invariant_upwinding_treatment.jl

```diff
@@ -0,0 +1,148 @@
+"""Upwinding treatment of Kinetic Energy Gradient and Divergence fluxes for the Vector Invariant formulation""" 
+abstract type AbstractUpwindingTreatment end
+
+struct OnlySelfUpwinding{A, U, V, U2, V2} <: AbstractUpwindingTreatment 
+    cross_scheme    :: A  # advection scheme for cross-reconstructed terms (in both divergence flux and KE gradient)
+    δU_stencil      :: U  # stencil used for assessing u-derivative smoothness
```

Comment:
If delta means "derviative" then maybe a partial is a better symbol. What direction is the derivative? If its the 2d gradient, then nabla could be better.